### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by exacs**
This PR

- Replaces the testing library from `tape` to `ava` since the latter allows us to create snapshot testing
- Creates three snapshot testing where we make sure that we create the same files that are going to be sent to Canvas. Files are part of the repo to be able to do the comparison.
- Eliminates a variable called `period` which was _only used for adding it to the `.csv` filenames_ and was confusing. I thought this app only handled certain periods!

Notes on the test in `test/integration/createEnrollmentsFile.test.js`:

The test was created to ensure that enrollments generated by this app are always the same. To do it, I decided to picked an old year (2017) and check that the app consistently generates the same file.

- Teachers, course responsibles and TAs are generated correctly and equally every day because they are internally defined per "course round"
- However examiners are defined in a course-level, meaning that when an examiner changes, it affects to the files generated for old course rounds of the same course.

Therefore is not possible to compare if examiners are the same always. It is possible to compare only teachers, course responsibles and TAs but it involves either a change in the code or more (in my opinion "too") advanced tests.

### CODE REVIEW

File: test/integration/createEnrollmentsFile.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const test = require('ava')
+const createCoursesFile = require('../../server/createCoursesFile')
+const createEnrollmentsFile = require('../../server/createEnrollmentsFile')
+const fs = require('fs')
+
+function removeId (line) {
+  return line.replace(/u\w{6}/i, 'uXXXXXX')
+}
+
+test.skip('Generate enrollments.csv files for 2017-1-3', async t => {
```
Comment (me97esn) [inline test/integration/createEnrollmentsFile.test.js:11]: Should this test be skipped? If so, should it still remain or be deleted?

File: package.json [hunk 2]
```diff
@@ -8,8 +8,8 @@
     "start": "NODE_ENV=production nodemon app.js",
     "start-dev": "cross-env NODE_ENV=development nodemon app.js",
     "format": "prettier-standard \"**/*.{js,md,yml,json}\"",
-    "test": "prettier-standard --lint --check \"**/*.{js,md,yml,json}\" && tape \"test/unit/**/*.js\" | tap-spec",
-    "test:integration": "tape \"test/integration/**/*.js\" | tap-spec",
+    "test": "prettier-standard --lint --check \"**/*.{js,md,yml,json}\" && ava \"test/unit/**/*.js\"",
+    "test:integration": "ava \"test/integration/**/*.js\" -v --timeout=60m",
```
Comment (me97esn) [inline package.json:12]: How did you generate the snapshot files? Can you add a npm script so we can re-generate them?

File: test/integration/createCoursesFile.test.js [hunk 3]
```diff
@@ -1,5 +1,27 @@
-const test = require('tape')
+require('dotenv').config()
```
Comment (me97esn) [inline test/integration/createCoursesFile.test.js:1]: Should these files be stored in the `integration` folder? Or should we create a `snapshot` folder for these kinds of tests instead? And run them with npm run test or npm run test:snapshot

File: test/integration/createEnrollmentsFile.test.js (hunk 1 above)
Comment (exacs) [reply test/integration/createEnrollmentsFile.test.js:11]: You can see the reason why is it skipped in the description of this PR. We can have a discussion on this

File: test/integration/createCoursesFile.test.js (hunk 3 above)
Comment (exacs) [reply test/integration/createCoursesFile.test.js:1]: I don't have any opinions on this. I would consider those tests as _some kind of_ integration/regression test.

However the snapshots themselves are in a directory called `snapshots` under `integration` so be careful to not have files under a directory called `snapshot/snapshots` which lead to confusions

File: test/integration/createEnrollmentsFile.test.js (hunk 1 above)
Comment (me97esn) [reply test/integration/createEnrollmentsFile.test.js:11]: Ok, read it now. In that case I say we remove it, let's start with the other tests and then we can see how they work first

**Pull request merged**